### ISSUE TITLE
Last move doesn't get analysis

### ISSUE THREAD

**Issue by barrtgt** [pnprog/goreviewpartner#121]

**Comment by pnprog** [comment]
Hi,
Can you share a sgf file where last move doesn't get analysis?  I cannot reproduce the issue.

**Comment by barrtgt** [comment]
Maybe I'm confused. The variation that is shown for the last move (the stone that isn't shown in the "Actual game", only a circle with the letter A) is an analysis after that move is played or the move before? It doesn't show up in the comments and it also doesn't show up in the converted csv.
![grp_test_csv](https://user-images.githubusercontent.com/34060320/54579440-b625f480-49c0-11e9-998b-3c73c0db7057.jpg)
![grp_test_sabaki](https://user-images.githubusercontent.com/34060320/54579498-ef5e6480-49c0-11e9-8f5f-b44218344fb2.jpg)
![grp_test_last](https://user-images.githubusercontent.com/34060320/54579551-2cc2f200-49c1-11e9-875d-887fdc868608.jpg)

**Comment by pnprog** [comment]
Yes you are somehow right, in fact, half the analysis is performed:

+ LeelaZero provide the alternative moves for the position
+ But the win rate of that last move is missing

The reason the win rate of the last move is missing if because GRP evaluates the win rate of move N by asking Leela Zero her best move win rate at N+1. GRP does that because quite often, the game move is not part of the best move proposed by Leela Zero.

In your example, the game move corresponds to Leela Zero third choice (variation C) but what if it is not considered by Leela Zero at all? For this reason, GRP asks Leela Zero to place that move on the board, then to play the following move with the opposite color. If the win rate of Leela Zero for move 8 is 45%, then GRP deduces that move 7 win rate is 100-45=55%. This also allow to have Leela Zero spend of her playout and that board position, so the evaluation is as accurate as possible. And since the following move has to be analyzed, it's not a problem.

So the win rate of move N is found when analyzing move N+1, which means the last move does not get a proper evaluation. It's usually not a problem, when the last move of a game is pass, or resign, but in your example, this is a problem. So maybe I should ask Leela to spend some time on the last last move even if it's not pass or resign.

> it also doesn't show up in the converted csv.

This also is an issue, I will fix that as well.
